### PR TITLE
#40 coin selection for wallet_send

### PR DESCRIPTION
**Pull request by k9ert**
There are still some CSS fixes to be done, though. Doesn't look that nice:

image

### CODE REVIEW

**Comment by ben-kaufman** [comment]
Hi @k9ert , great feature!
If you want I could PR some UI fixes to your branch.

**Comment by k9ert** [comment]
I'm definitely not very good in CSS and it would be cool if you can help here. I've invited you to my fork. You can directly commit in there! Thanks a lot!

**Comment by ben-kaufman** [comment]
Great, just pushed a commit with a few fixes, the UI should look fine now. Let me know if it's working well for you.

**Comment by k9ert** [comment]
Checked Ben's improvements and looks good to me. I'd merge that with ACK from @ben-kaufman and @stepansnigirev . Stepan is currently quite busy and can probably only roughly review this, so would be cool to have also Ben to confirm this.

File: src/cryptoadvance/specter/templates/wallet_send.html [hunk 1]
```diff
@@ -390,7 +434,58 @@ <h2>Sign Transaction</h2>
...
 		});
 {% endif %}
+</script>
+
+<script>
+	var spend_coins = new Vue({
+		el: '#spend_coins',
+		delimiters: ["[[", "]]"],
+		data: {
+			unspents: {{ wallet.cli.listunspent()|tojson }},
```
[the remark this answers is not in the document]
Comment (stepansnigirev) [reply src/cryptoadvance/specter/templates/wallet_send.html:444]: I agree, unconfirmed utxo should be in the list as well

**Comment by stepansnigirev** [comment]
When I click on "Create transaction" nothing happens even though I selected enough coins.

**Comment by ben-kaufman** [comment]
@stepansnigirev this is because of this issue: https://github.com/cryptoadvance/specter-desktop/pull/96#discussion_r402414447 The button is disabled when it should be enabled.
Once the condition is fixed it should work well.

File: src/cryptoadvance/specter/controller.py [hunk 2]
```diff
@@ -411,15 +411,16 @@ def wallet_send(wallet_alias):
...
+            print("selected coins: {}".format(selected_coins))
             if 'dynamic' in request.form.get('fee_options'):
                 fee_rate = float(request.form.get('fee_rate_dynamic'))
             else:
                 if request.form.get('fee_rate'):
                     fee_rate = float(request.form.get('fee_rate'))
 
             try:
-                psbt = wallet.createpsbt(address, amount, subtract=subtract, fee_rate=fee_rate, fee_unit=fee_unit)
+                psbt = wallet.createpsbt(address, amount, subtract=subtract, fee_rate=fee_rate, selected_coins=selected_coins)
```
[the remark this answers is not in the document]
Comment (k9ert) [reply src/cryptoadvance/specter/controller.py:423]: Ups, yes, good point.

**Comment by ben-kaufman** [comment]
@stepansnigirev This PR should be ready now.

**Comment by stepansnigirev** [comment]
Checked, works. Merging!

**Pull request merged**